components/map-menu/actions.js

```diff
@@ -10,58 +10,64 @@ export const setMenuSettings = createAction('setMenuSettings');
 
 export const getLocationFromSearch = createThunkAction(
   'getLocationFromSearch',
-  ({ search, token, lang }) => (dispatch) => {
-    dispatch(setMenuLoading(true));
-    if (search) {
-      fetchGeocodeLocations(search, lang, token)
-        .then((locations) => {
-          if (locations?.length) {
-            dispatch(setLocationsData(locations));
-          } else {
-            dispatch(setLocationsData([]));
-          }
-          dispatch(setMenuLoading(false));
-        })
-        .catch(() => {
-          dispatch(setMenuLoading(false));
-        });
+  ({ search, token, lang }) =>
+    (dispatch) => {
+      dispatch(setMenuLoading(true));
+      if (search) {
+        fetchGeocodeLocations(search, lang, token)
+          .then((locations) => {
+            if (locations?.length) {
+              dispatch(setLocationsData(locations));
+            } else {
+              dispatch(setLocationsData([]));
+            }
+            dispatch(setMenuLoading(false));
+          })
+          .catch(() => {
+            dispatch(setMenuLoading(false));
+          });
+      }
     }
-  }
 );
 
 export const handleClickLocation = createThunkAction(
   'handleClickLocation',
-  ({ center, bbox: featureBbox, ...feature }) => (dispatch) => {
-    if (featureBbox) {
-      dispatch(setMapSettings({ canBound: true, bbox: featureBbox }));
-    } else {
-      dispatch(
-        setMapSettings({ center: { lat: center[1], lng: center[0] }, zoom: 12 })
-      );
+  ({ center, bbox: featureBbox }) =>
+    (dispatch) => {
+      if (featureBbox) {
+        dispatch(setMapSettings({ canBound: true, bbox: featureBbox }));
+      } else {
+        dispatch(
+          setMapSettings({
+            center: { lat: center[1], lng: center[0] },
+            zoom: 12,
+          })
+        );
+      }
+
```

Comment:
Nice! I'm glad removing the analysis modal wasn't complicated, @willian-viana . 😅